### ISSUE TITLE
Make a repository of OpenCv builds for Appropriate gocv versions (Atleast for upcoming versions of gocv).

### ISSUE THREAD

**Issue by SakthiMahendran** [hybridgroup/gocv#925]
I suffered a lot by trying to make gocv work. But still I can't !.

The build script that is used to build OpenCv for gocv just failed.

This seems like a comman problem.
That most people face but I found only one solution in online which suggests to build OpenCv manually. 

but the problem is I am mostly used Java and Python.

And have no Idea on cmake and other gnu and gcc stuff and most of the developers too.

Even though I tried to build OpenCv manually. It just failed.

I am frustrated after 2 weeks of repeating the same stuff.

Now I am wondering why don't we just include a appropriate version of OpenCv build for gocv in this repository or in new repository atleast for upcoming gocv versions.

If this happens it will be great help for poor developer like me.

Thank you 😊.

**Comment by TinkerTurtle** [comment]
Have you tried the instructions here? https://gocv.io/getting-started/linux/

The Makefile lists versions at the top, and currently it even works if you substitute the values for latest versions of OpenCV (4.5.5) and Go (1.17.6).

**Comment by SakthiMahendran** [comment]
> Have you tried the instructions here? https://gocv.io/getting-started/linux/
> 
> The Makefile lists versions at the top, and currently it even works if you substitute the values for latest versions of OpenCV (4.5.5) and Go (1.17.6).

I am not sure. I tried a lot and frustrated.

Why should we force a developer to build OpenCv on their own. Because it takes lots of time and difficult especially for beginners. 

So we can just build appropriate version of OpenCv for gocv (atleast for upcoming versions of gocv) and maintain an repository for it.

Hope you understand.

**Comment by softexpert** [comment]
I agree with the request - I built a project locally and when I need to deploy it I need to build at the destination an identical copy of the OpenCV version as the one used for development.
Not everyone is able to use Docker containers ; having more alternatives is highly desirable.

**Comment by SakthiMahendran** [comment]
> I agree with the request - I built a project locally and when I need to deploy it I need to build at the destination an identical copy of the OpenCV version as the one used for development.
> Not everyone is able to use Docker containers ; having more alternatives is highly desirable.

This is a good point.

If we want to distribute our application to other platforms where we can get the OpenCv build for that particular platform?.

It will be a huge pain right.

So it is very important to maintain an OpenCv build for gocv.

**Comment by deadprogram** [comment]
Please see the tags I have created here matching specific versions of OpenCV 
https://github.com/hybridgroup/gocv/tags

For example, this is the version of GoCV to match OpenCV 4.5.2: 
https://github.com/hybridgroup/gocv/releases/tag/opencv-4.5.2

Hopefully this can provide some clarity about version compatibility.
